docs/en/custom-elements/supplemental.md

```diff
@@ -0,0 +1,239 @@
+
+```html
+<ref-email-signup id="emailSignup" title="Subscribe to Our Newsletter"></ref-email-signup>
+```
+
+### Events
+
+All widget properties that start with `on` and have a function signature are exposed as events. The emitted event is the lower cased part of the property name without the "on." Listening for the onSignUp property (`{ onSignUp?: () => void }`) would look like this.
+
+```html
+<script>
+    emailSignup.addEventListener('signup', () => {
+        // start confetti animation 🎉
+    });
+</script>
+```
+
+### Properties
+
+All non-event widget properties are automatically created as properties on the custom element. These properties can be fully controlled by JavaScript. In the email subscription widget, both the title and signUpBonusDate properties can be set programmatically as properties.
+
+```html
+<script>
+        const now = new Date();
+        emailSignup.signUpBonusDate = new Date(
+            now.getFullYear(),
+            now.getMonth(),
+            now.getDate() + 1
+        );
+    </script>
+```
+
+# Custom Element Slots
+
+Some Dojo widgets utilize child objects rather than child widgets. These child objects are supported as custom elements by using "slots." A `slot` attribute is added to a child of the custom element, and the child gets passed in as the value of the slot.
+
+The most common child object scenarios are supported by Dojo custom elements.
+
+## Named Children
+
+> Widget.tsx
+
+```tsx
+export interface WidgetChildren {
+	foo: RenderResult;
+}
+```
+
+```tsx
+<Widget>
+	{{
+		foo: <Label>My Label</Label>
+	}}
+</Widget>
+```
+
+> index.html
+
+```html
+<dojo-widget>
+  <dojo-label slot="foo">My Label</dojo-label>
+</dojo-widget>
+```
+
+## Named Child Renderers
+
+> Widget.tsx
+
+```tsx
+export interface WidgetChildren {
+	foo: () => RenderResult;
+}
+```
+
+```tsx
+<Widget>
+	{{
+		foo: () => <Label>My Label</Label>
+	}}
+</Widget>
+```
+
+> index.html
+
+```html
+<dojo-widget>
+  <dojo-label slot="foo">My Label</dojo-label>
+</dojo-widget>
+```
+
+## Named Array of Children
+
+> Widget.tsx
+
+```tsx
+export interface WidgetChildren {
+	foo: RenderResult[];
+}
+```
+
+```tsx
+<Widget>
+	{{
+		foo: [<Label>Label 1</Label>, <Label>Label 2</Label>]
+	}}
+</Widget>
+```
+
+## Named Array of Child Renderers
+
+> Widget.tsx
+
+```tsx
+export interface WidgetChildren {
+	foo: (() => RenderResult)[];
+}
+```
+
+```tsx
+<Widget>
+	{{
+		foo: [() => <Label>Label 1</Label>, () => <Label>Label 2</Label>]
+	}}
+</Widget>
+```
+
+> index.html
+
+```html
+<dojo-widget>
+  <dojo-label slot="foo">Label 1</dojo-label>
+  <dojo-label slot="foo">Label 2</dojo-label>
+</dojo-widget>
+```
+
+## Named Child Renderers with Arguments
+
+> Widget.tsx
+
+```tsx
+export interface WidgetChildren {
+	foo: (active: boolean) => RenderResult;
+}
+```
+
+```tsx
+<Widget>
+	{{
+		foo: (active) => <Label active={active}>My Label</Label>
+	}}
+</Widget>
+```
+
+> index.html
```

Comment:
Are we going to show how to use the arguments passed to the child renderer?

Reply:
Oh! Good catch. meant to include that.